tests/_require_all.rs

```diff
@@ -0,0 +1,8 @@
+#![cfg(not(all(
+    feature = "os-poll",
+    feature = "os-util",
+    feature = "tcp",
+    feature = "udp",
+    feature = "uds"
+)))]
+compile_error!("run main Mio tests with `--all-features`");
```

Comment:
Don't we also want to support tests with partial features enabled? Maybe not now but in the future?

Reply:
Possibly! This works well for now, but should be easy to change in the future as our support for tests do. I don't want to worry about that now with this change though.

Reply:
Tracked in #1214.